CODING_STANDARDS.md

```diff
@@ -0,0 +1,263 @@
+Rather than:
+
+```cpp
+if (condition)
+{
+   <logic>
+}
+return error_code;
+```
+
+Use:
+
+```cpp
+if (!condition)
+  return error_code;
+<logic>
+return success;
+```
+
+### File names
+
+File names should be lower case with underscore word separators.
+C file names use the `.c` extension, C++ files use the `.cc` extension and header files use the `.h` extension.
+
+### Language standards
+
+For pure-C files we use C99 and for C++ we use C++11.
+The code need to be able to compile on multiple platforms using different compilers (for example: Windows / Linux, x86_64 / ARM).
+
+### Line lengths
+
+Lines should be no more than 80 characters.
+
+### Comments
+
+Single line / inline code comments can use the double slash (`//`) style of coding, whereas multi-line code comments should use `/*` as a start and `*/` at the end, with the text indented by 2 spaces, for example:
+
+```cpp
+/*
+  This is a multi-line code comment.
+  It has an indentation of two spaces.
+*/
+```
+
+### Variables classes, and functions
+
+Variables and functions should be descriptive and in "snake case", for example:
+
+```cpp
+void my_function(uint16 variable_name)
+{
+```
+
+Class names should also be "snake case" but should start with an upper-case character.
+Such as this:
+
+```cpp
+class Buffered_logs
+{
+```
+
+Assignments should not have a space on the left side of the equals. For example:
+
+```cpp
+a= 1;  // Correct
+a = 1; // Incorrect for the server code,
+       // ok for Storage Engines if they use it (aka Connect)
+```
+
+Please do not write conditions like this:
+
+```cpp
+if (0 == *error_code)
+```
+
+Please do this instead:
+
+```cpp
+if (*error_code == 0)
+// Or even better
+if (!*error_code)
+```
+
+### Constant integers
+
+Constant integers that are used to define elements such as buffer sizes should be defined rather than used directly.
+This is because the integer could change and uses of it could be missed.
+For example:
+
+```cpp
+char *buffer= my_malloc(PSI_INSTRUMENT_ME, 1024, MYF(MY_WME));
+
+snprint(buffer, 1024, "%d: %s", integer, text);
+```
+
+Could become:
+
+```cpp
+constexpr int buffer_size= 1024;
+char *buffer= my_malloc(PSI_INSTRUMENT_ME, buffer_size, MYF(MY_WME));
+
+snprint(buffer, buffer_size, "%d: %s", integer, text);
+```
+
+Alternatively the integer can be defined using an `enum` or `#define`.
+
+### Spacing
+
+#### Whitespace
+
+* Lines should not have any trailing whitespace.
+* There should not be any trailing blank lines at the end of a file.
+* Line endings are POSIX style (`\n`).
+* Two spaces for each indentation level, not tabs.
+
+#### Pointers
+
+The `*` of a pointer should be on the side of the variable name such as:
+
+```cpp
+void my_function(THD *thd)
+{
+```
+
+As yet there is no standard as to whether the `*` in a casting should have a space or not.
+Both of these are valid:
+
+```cpp
+name= (const char*) db_name;
+name= (const char *) db_name;
+```
+
+#### Equals
```

Comment:
combine this with "assignment" above

Reply:
I think that `(const char*) db_name;` should be written without extra space after closing parenthesis, since type convertion is an unary operator

Reply:
> I think that `(const char*) db_name;` should be written without extra space after closing parenthesis, since type convertion is an unary operator

Since our usage here is mixed I'll remove that part of the specification for now until it can be decided as to what is correct.